src/main/java/bio/overture/ego/service/TokenService.java

```diff
@@ -282,25 +281,33 @@ public String generateAppToken(Application application) {
     val tokenClaims = new AppTokenClaims();
     tokenContext.setScope(permissionNames);
     tokenClaims.setIss(ISSUER_NAME);
-    tokenClaims.setValidDuration(JWT_DURATION);
+    tokenClaims.setValidDuration(APP_JWT_DURATION);
     tokenClaims.setContext(tokenContext);
     return getSignedToken(tokenClaims);
   }
 
   public boolean isValidToken(String token) {
     Jws<Claims> decodedToken = null;
+    val tokenKey =
+        tokenSigner
+            .getKey()
+            .orElseThrow(() -> new InternalServerException("Internal issue with token signer."));
     try {
-      decodedToken = Jwts.parser().setSigningKey(tokenSigner.getKey().get()).parseClaimsJws(token);
+      decodedToken = Jwts.parser().setSigningKey(tokenKey).parseClaimsJws(token);
     } catch (JwtException e) {
       log.error("JWT token is invalid", e);
     }
     return (decodedToken != null);
   }
 
   public Jws<Claims> validateAndReturn(String token) {
-    Jws<Claims> decodedToken = null;
+    Jws<Claims> decodedToken;
```

Comment:
refactor so that `Optional<Jwa<Claims>> processValidToken(@NonNull String token)` does all the meat of the work, and then
```
public boolean isValidToken(String token){ 
   return processValidToken(token).isPresent(); 
}

public Jws<Claims> validateAndReturn(String token){
     return processValidToken(token).orElseThrow(() -> {
        log.error("JWT token was null when trying to validate and return.");
        return new ForbiddenException("Authorization is required for this action.");
     });
}
```

Reply:
my man, that is legit